pkg/controller/utils/auth.go

```diff
@@ -99,12 +102,36 @@ func GetIDPSecret(ctx context.Context, client client.Client, authentication *ope
 		if len(data) == 0 {
 			return nil, fmt.Errorf("%s is a required field for secret %s/%s", field, secret.Namespace, secret.Name)
 		}
+	}
 
-		if field == render.BindDNSecretField {
-			if _, err := ldap.ParseDN(string(data)); err != nil {
-				return nil, fmt.Errorf("secret %s/%s field %s: should have be a valid LDAP DN", common.OperatorNamespace(), secretName, field)
-			}
+	// Validate LDAP fields.
+	bindDN := secret.Data[render.BindDNSecretField]
+	bindPW := secret.Data[render.BindPWSecretField]
+	if len(bindDN) > 0 && len(bindPW) > 0 {
+		if _, err := ldap.ParseDN(string(bindDN)); err != nil {
+			return nil, fmt.Errorf("secret %s/%s: should have be a valid LDAP DN", common.OperatorNamespace(), secretName)
+		}
+
+		// When Dex starts, it uses os.ExpandEnv to set up its configuration inside a json formatted string. If we validate
+		// the input in the operator, the user will catch the configuration issue sooner.
+		const config = `{"bindDN": "$BIND_DN", "bindPW": "$BIND_PW"}`
+		data := []byte(os.Expand(config, func(s string) string {
+			return map[string]string{
+				"BIND_DN": string(bindDN),
+				"BIND_PW": string(bindPW),
+			}[s]
+		}))
+		if err := json.Unmarshal(data, &map[string]any{}); err != nil {
+			return nil, fmt.Errorf("secret %s/%s should have properly escaped data for bindDN and bindPW", common.OperatorNamespace(), secretName)
```

Comment:
Nit, would the error message be better worded stating just that it's not escaped "Fields bindDN and bindPW in secret %s/%s are not properly escaped"?